src/software/SfM/main_SfMInit_ImageListing.cpp

```diff
@@ -127,6 +127,176 @@ std::pair<bool, Vec3> checkPriorWeightsString
+	{
+		std::cerr << "Unable to open landmark file for reading: " << sFilename << std::endl;
+		return (false);
+	}
+
+    // If there are already control points, clear them
+    if(!sfm_data.control_points.empty())
+    {
+        std::cout << "Removing " << sfm_data.control_points.size() << " existing landmarks." << std::endl;
+    }
+
+	// Read in each landmark
+	std::string sLine;
+	std::string landmark_name;
+	double x, y, z;
+	int num_control_points;
+	while (std::getline( theFile, sLine))
+	{
+		// Skip empty
+		if (sLine.empty())
+		{
+			continue;
+		}
+
+		// Parse string of form: landmark_name,x,y,z,#control_points_to_follow
+		auto tokens = parseDelimitedString(sLine, ",");
+		if (tokens.size() < 5)
+		{
+			std::cerr << "Unrecognized format in landmarks file for line: " << sLine << std::endl;
+			continue;
+		}
+		landmark_name = tokens[0];
+		x = ::atof(tokens[1].c_str());
+		y = ::atof(tokens[2].c_str());
+		z = ::atof(tokens[3].c_str());
+		num_control_points = ::atoi(tokens[4].c_str());
+		if ((num_control_points < 0) || (num_control_points > sfm_data.GetViews().size()))
+		{
+			std::cerr << "Unexpected control point count of " << num_control_points << " for landmark " << landmark_name << std::endl;
+			continue;
+		}
+
+		// Start landmark
+		Landmark landmark;
+		landmark.X = Vec3(x,y,z);
+
+		// Read in each control point
+		for (int i = 0; i < num_control_points; ++i)
+		{
+			// Read in the line string
+			if (!std::getline(theFile, sLine))
+			{
+				std::cerr << "Unexpected end of file reading control points for landmark " << landmark_name << std::endl;
+				break;
+			}
+			// Read in the pixel coordinates and image name
+			auto tokens = parseDelimitedString(sLine, ",");
+			if ( tokens.size() < 3)
+			{
+				std::cerr << "Unrecognized format in control point " << i << " for landmark " << landmark_name << std::endl;
+				break;
+			}
+			std::string image_name = tokens[0];
+			double pix_x = ::atof(tokens[1].c_str());
+			double pix_y = ::atof(tokens[2].c_str());
+
+			// Determine view ID from image filename. Should be only point on this view for the landmark.
+			IndexT view_id = 0;
+			bool view_id_valid = false;
+			for (const auto& view : sfm_data.GetViews())
+			{
+				// Extract filename part of entire path
+				const auto& view_full_filename = view.second->s_Img_path;
+				auto last_slash_pos = view_full_filename.find_last_of("\\/");
+				std::string view_base_filename;
+				if (view_full_filename.npos == last_slash_pos)
+				{
+					view_base_filename = view_full_filename;
+				}
+				else
+				{
+					view_base_filename = view_full_filename.substr(last_slash_pos + 1);
+				}
+				
+				// Remove extension
+				auto last_period_pos = view_base_filename.find_last_of('.');
+				if (view_base_filename.npos != last_period_pos)
+				{
+					view_base_filename = view_base_filename.substr(0, last_period_pos);
+				}
+
+				// Compare the strings
+				if (0 == stricmp(view_base_filename.c_str(), image_name.c_str()))
```

Comment:
Non-portable use of stricmp ?

Reply:
It looks like CPL uses a construct like below for a POSIX or Windows version of caseless compare:

`#  if defined(WIN32)
#    define STRCASECMP(a,b)         (stricmp(a,b))
#    define STRNCASECMP(a,b,n)      (strnicmp(a,b,n))
#  else
/** Alias for strcasecmp() */
#    define STRCASECMP(a,b)         (strcasecmp(a,b))
/** Alias for strncasecmp() */
#    define STRNCASECMP(a,b,n)      (strncasecmp(a,b,n))
#  endif
`
Of course, until we actually build for another platform this doesn't really matter, so probably no reason to change it until then.

Reply:
I have no idea why the 'Insert Code' made the font so big for some lines of that chunk of code.

Reply:
I added a macro definition
`
#ifdef _MSC_VER
#define strcasecmp(x, y) _stricmp(x, y)
#endif
`